add-on/src/lib/ipfs-client/index.js

```diff
@@ -1,7 +1,7 @@
 'use strict'
 
-const embedded = require('./embedded')
 const external = require('./external')
+const embedded = require('./embedded')
```

Comment:
Some feedback from the last IPFS all hands was that `embedded` might confuse devs to think that know there is a `window.ipfs` in every page. Is that one of the goals too? It would be a good feature to have (optionally)

Reply:
I think that would be simple to add. Is the idea that uses would write scripts against `window.ipfs`  but not bundle it with their script? It sounds a little bit like those browser extensions that add jquery to every page, so you can noodle around in the dev console. 

Do people have a use-case in mind? T feel like we should be directing devs to the js-ipfs examples, to show them how to bundle it.

Reply:
The difference between adding jquery and ipfs is that on one you would be just packing more code, the second you are opening more ports and doing more crypto which are both expensive operations.

The Metamask team has been injecting Web3 to every page to let devs use Ethereum directly if WebPages are there.

I feel this should be one of the options for the extension and until we have concrete use cases, just refer to an example on how to use it to seed it to developers.

Reply:
Lots of benefits for embedding an already running IPFS node in webpages, too many to list here, but for starter, it makes startup times a lot better as nothing has to initialized, just start using it directly (node in background is already running so!). Performance of loading pages and loading content will be faster as well. Things can be shared via MFS between applications, as long as they agree on a standard structure. ID can follow between applications as well. Just a few reasons.

Reply:
Very cool! I like the idea of making ipfs available to webapps and seeing what people do with it.

Reply:
See https://github.com/ipfs/ipfs-companion/issues/330, I created it to track `window.ipfs`  effort :)

Reply:
This! https://github.com/VictorBjelkholm/js-ipfs-in-the-browser